xsplice: Relax .xsplice.funcs checks

old_addr is not filled in by xsplice-build if dynamic linking is used so
don't fail if it is not set.
old_size and new_size are not used, so don't bother checking them.

diff --git a/xen/common/xsplice.c b/xen/common/xsplice.c
index 3050ad4..11e0e30 100644
--- a/xen/common/xsplice.c
+++ b/xen/common/xsplice.c
@@ -720,7 +720,7 @@ static int find_special_sections(struct payload *payload,
 
         f = &(payload->funcs[i]);
 
-        if ( !f->new_addr || !f->old_addr || !f->old_size || !f->new_size )
+        if ( !f->new_addr )
             return -EINVAL;
 
         for ( j = 0; j < 8; j ++ )
